File: api/app/modules/jdmatch/utils/is_jd_link_or_description.py
import re


def is_jd_link_or_description(text: str) -> bool:
    url_regex = r"^(https?://)?([\w-]+\.)+[\w-]+(/[\w\-._~:/?#\[\]@!$&'()*+,;=]*)?$"
    return bool(re.match(url_regex, text.strip(), re.IGNORECASE))

File: shared/ai/jd/isJdLinkOrDescription.ts
export default function isJdLinkOrDescription(text: string): boolean {
  const urlRegex = /^(https?:\/\/)?([\w-]+\.)+[\w-]+(\/[\w\-._~:/?#[\]@!$&'()*+,;=]*)?$/i;
  return urlRegex.test(text.trim());
}
